packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx

```diff
@@ -103,7 +103,7 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
       isPrimaryButtonDisabled!,
       checked!,
       !!this.state.menuProps,
-      this.props.split) : getBaseButtonClassNames(styles!, className!,
+      this.props.split) : getBaseButtonClassNames(this.props.classNames!, styles!, className!,
```

Comment:
@dzearing  @christiango To get around the issue where I needed to send merge all classNames (because by only sending the className, selectors would be missing reference to svg) I added classNames as a prop. I am still not sure this is the best idea but I am open to suggestions.

Reply:
Hmm, I'm not sure I follow the scenario. What is it that you need to do and what exactly goes in classNames in the swatch color picker scenario?

Reply:
Given that we apply selector effects (hover, focus, active) to the svg through the parrent (button element), when we were  sending the className to use on the GridCell, it is merged again with ActionButtonStyles (given that a gridCell renders a CommandButton). Through this second merge, the svg className was unknown so the selectors reference were missing svg's className correct name.

Reply:
Hmm, I'd wait for david's guidance on how to proceed here. Seems tricky

Reply:
Sure! I spoke to him earlier and he said he'd take a look tonight :)